compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -4111,6 +4111,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return false;
         }
         let trait_def_id = self.tcx.parent(item_def_id);
+        if !self.tcx.is_trait(trait_def_id) {
```

Comment:
Could you rename this function to `detect_and_explain_multiple_crate_versions_of_trait_item`. Specifically, adding "`of_trait_item`" to make it clear what this is doing?

r=me afterwards